### ISSUE TITLE
Javascript Preprocessor/Language

### ISSUE THREAD

**Issue by SevereOverfl0w** [SevereOverfl0w/Unicorn-UI-Kit#3]
Coffeescript, Typescript, Flow, 6to5?

I think 6to5 is great, as it's essentially an ES6 polyfill. Makes us all future proof. ES6 contains a lot of Coffeescript improvements.

**Comment by SevereOverfl0w** [comment]
With further reading, I have decided I quite like the look of Typescript & ES6. I think they make a great combination, the static typing will help us catch errors early on, and ES6 gives us a LOT more flexibility. If we use 6to5, we can be backwards compatible with ES5 browsers.

**Comment by SevereOverfl0w** [comment]
Definitely going typed. @connor4312 suggested not using Facebook's Flow due to it's OCaml dependency. Typescript is pretty much the only contender as a result.

I'm not overly fond of Typescript basically including an ES6 preprocessor. But it's our only option at this point really, if we want to be typed. I like the idea, as it will make it easier to catch bugs in our code.

tl;dr We're gonna use typescript.

**Comment by connor4312** [comment]
:+1:
